pyvista/utilities/reader.py

```diff
@@ -2068,6 +2072,85 @@ class SegYReader(BaseReader):
     _class_reader = staticmethod(_vtk.lazy_vtkSegYReader)
 
 
+class _GIFReader:
+    """Simulate a VTK reader for GIF files."""
+
+    _data_object = None
+    _observers: List[Union[int, Callable]] = []
+    _n_frames = 0
+    _current_frame = 0
+
+    def SetFileName(self, filename):
+        """Needed for VTK-like compatibility with BaseReader."""
+        self._filename = filename
+
+    def UpdateInformation(self):
+        """Needed for VTK-like compatibility with BaseReader."""
+        pass
+
+    def AddObserver(self, event_type, callback):
+        """Needed for VTK-like compatibility with BaseReader."""
+        self._observers.append([event_type, callback])
+
+    def RemoveObservers(self, *args):
+        """Needed for VTK-like compatibility with BaseReader."""
+        pass
+
+    def GetProgress(self):
+        return self._current_frame / self._n_frames
+
+    def UpdateObservers(self, event_type):
+        for event_type_allowed, observer in self._observers:
+            if event_type_allowed == event_type:
+                observer(self, event_type)
+
+    def Update(self):
+        """Read the GIF and store internally to `_data_object`."""
+        from PIL import Image, ImageSequence
+
+        img = Image.open(self._filename)
+        self._data_object = pyvista.UniformGrid(dims=(img.size[0], img.size[1], 1))
+
+        # load each frame to the grid (RGB since gifs do not support transparency
+        self._n_frames = img.n_frames
+        for i, frame in enumerate(ImageSequence.Iterator(img)):
+            self._current_frame = i
```

Comment:
For a future implementation, one important thing to consider now is the array data name.  For one-frame-at-a-time, would we use `frame` without an index?  Another thought is `image-frame{i}` and `image`.

Reply:
> This implementation reads all data at once. For large gifs, I wonder if there should be an option to read in frames one at a time using TimeReader. This could be a separate PR. Just raising this in case in spurs thoughts on implementation here.

Let's see how many people actually use this. I'll use it for displaying simple animations, but I doubt many others will use this.